src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,387 @@
+         *
+         * @return Returns the vendor name
+         */
+        const std::string getVendor(Utils::Registry& registry)
+        {
+            std::string vendor;
+
+            for (const auto& folder : registry.enumerate())
+            {
+                try
+                {
+                    Utils::Registry fileReg(m_key, m_userId + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + FILE_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                    vendor = searchPublisher(fileReg);
+
+                    if (vendor.size() == 0)
+                    {
+                        Utils::Registry urlReg(m_key, m_userId + "\\"  + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + URL_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                        vendor = searchPublisher(urlReg);
+                    }
+                }
+                catch (...)
+                {
+                }
+            }
+
+            return vendor;
+        }
+
+        /*
+         * @brief Search the name application in the cache registry.
+         *
+         * @param[in] nameApp Abbreviation of the name application to look for in the cache registry.
+         * @param[in] nameKey Name key where is the name application
+         *
+         * @return Return the name application.
+         */
+        const std::string searchNameFromCacheRegistry(const std::string& nameApp, const std::string& nameKey)
+        {
+            std::string registry;
+            std::string name;
+
+            // Looking for the named folder same like name app
+            for (const auto& folder : m_cacheReg)
+            {
+                if (folder.find(nameApp) != std::string::npos)
+                {
+                    registry = folder;
+                    break;
+                }
+            }
+
+            if (registry.size() != 0 )
+            {
+                name = searchKeyOnSubRegistries(m_userId + "\\" + CACHE_NAME_REGISTRY + "\\" + registry, nameKey);
+            }
+
+            return name;
+        }
+
+        /*
+         * @brief Search the key on the registry and sub-registries.
+         *
+         * @param[in] path Start path where you want to search.
+         * @param[in] key Name key to look for
+         *
+         * @return Return the value of the key.
+         */
+        const std::string searchKeyOnSubRegistries(const std::string& path, const std::string& key)
+        {
+            std::string value;
+            Utils::Registry registry(m_key, path);
+
+            if (!registry.string(key, value))
+            {
+                for (const auto& folder : Utils::Registry(m_key, path).enumerate())
+                {
+                    std::string tempPath { path + "\\" + folder };
+                    value = searchKeyOnSubRegistries(tempPath, key);
+
+                    if (value.size())
```

Comment:
Remember to check explicitly return values unless are booleans
```suggestion
                    if (!value.empty())
```